client/src/pages/login/login.js

```diff
@@ -25,8 +26,19 @@ const Login = ({ login, isAuthenticated, alert, toggleGuestMode, guestMode }) =>
   const onSubmit = async (e) => {
     e.preventDefault()
 
-    if (guestMode) toggleGuestMode()
-    login(email, password)
+    if (guestMode) 
+      toggleGuestMode()
+    
+    if (email && password) {
+      let info = await login(email, password)
+      info.message 
+      ? info.code == 'ERR_BAD_RESPONSE'
+        ? showToast('server error, connection failed', 'error')
+        : showToast('invalid username or password', 'warn')
+      : (function(){})()
```

Comment:
this block is super messy. Nested ternaries are definitely a code smell and aren't readable, especially when one of the default conditions is an empty function? You shouldn't need that. Also I don't love the idea of using the word "info" to describe anything, response is much more conventional, especially since we're checking for a "bad response" return anyways.

Reply:
fair enough, I've been using a lot of nested ternary stuff in my own code but good to know it's not a widely accepted norm. 
Yeah the empty function was there to prevent syntax errors -- for a while I rendered a 'success' message but decided to take it out so put that there as a "do nothing" instead
updated to an `if` check and then put the un-nested ternary logic in there 👍